app/action/itinerary-search-action.coffee

```diff
@@ -1,6 +1,57 @@
+last = require 'lodash/array/last'
 xhrPromise = require '../util/xhr-promise'
 config     = require '../config'
 
+create_wait_leg = (start_time, duration, point, placename) ->
+  leg =
+    mode: "WAITING"
+    routeType: null # non-transit
+    route: ""
+    duration: duration
+    startTime: start_time
+    endTime: start_time + duration
+    legGeometry: {points: [point]}
+    from:
+      lat: point[0] * 1e-5
+      lon: point[1] * 1e-5
+      name: placename
+    intermediateStops: []
+  leg.to = leg.from
+  return leg
+
+add_waiting_legs = (data) ->
+  for itinerary in data.plan?.itineraries or []
+    new_legs = []
+    time = itinerary.startTime # tracks when next leg should start
+    for leg in itinerary.legs
+      wait_time = leg.startTime - time
+      time = leg.endTime # next leg should start when this one ended
+
+      # If there's unaccounted time before a leg, add a waiting leg
+      if wait_time > 1000  # OTP often marks time between walk and vehicle as 1000ms
+        # OTP starts walking legs as late as possible,
```

Comment:
Not true?
![screen shot 2015-10-26 at 16 07 01](https://cloud.githubusercontent.com/assets/698100/10730932/9cbd737e-7bfb-11e5-81fb-bded6c7389ff.png)